src/query/interpreter.cpp

```diff
@@ -4964,33 +4971,36 @@ PreparedQuery PrepareCreateSnapshotQuery(ParsedQuery parsed_query, bool in_expli
     if (!maybe_path) {
       switch (maybe_path.error()) {
         case storage::InMemoryStorage::CreateSnapshotError::ReachedMaxNumTries:
-          spdlog::warn("Failed to create snapshot. Reached max number of tries. Please contact support");
+          spdlog::warn("Failed to create snapshot. {}. Please contact support.",
+                       storage::InMemoryStorage::CreateSnapshotErrorToString(maybe_path.error()));
           break;
         case storage::InMemoryStorage::CreateSnapshotError::AbortSnapshot:
-          throw utils::BasicException("Failed to create snapshot. The current snapshot needs to be aborted.");
+          throw utils::BasicException("Failed to create snapshot. {}.",
+                                      storage::InMemoryStorage::CreateSnapshotErrorToString(maybe_path.error()));
         case storage::InMemoryStorage::CreateSnapshotError::AlreadyRunning:
-          throw utils::BasicException("Another snapshot creation is already in progress.");
         case storage::InMemoryStorage::CreateSnapshotError::NothingNewToWrite:
-          throw utils::BasicException("Nothing has been written since the last snapshot.");
+          throw utils::BasicException("{}", storage::InMemoryStorage::CreateSnapshotErrorToString(maybe_path.error()));
```

Comment:
I think BasicException accepts passing string as arg, you shouldn't need formatting {}